R/importSNV.TCGA.R

```diff
@@ -0,0 +1,156 @@
+# importNet.TCGA.R
+
+#' Import MAF files from source and coverts them to rSNV files.
+#'
+#' \code{importSNV.TCGA} imports maf, creates rSNV, returns rSNV
+#'
+#' @section MAF file:...
+#'
+#' @param fMAF vector of MAF file names
+#' @param rSNV new rSNV file name to be created or the one to be merged with
+#' @param silent Controls whether output to console should be suppressed. FALSE
+#'   by default.
+#' @param noLog Controls whether writing the result to the global logfile is
+#'   enabled. TRUE by default.
+#' @return rSNV file name containg data from all MAF files
+#'
+#' @family
+#'
+#'   ## @seealso \code{\link{fastMap}} fastMap() is used internally to map ENSP
+#'   IDs to gene symbols.
+#'
+#'   ## @examples ## \dontrun{ ## importNet.STRING(IN, OUT) ## }
+#' @export
+importSNV.TCGA <- function(fMAF,
+                             rSNV,
+                             silent = FALSE,
+                             noLog = TRUE) {
+
+    # ==== PARAMETERS ==========================================================
+
+    # NL <- .PlatformLineBreak()
+    # cutoffTypes <- c("xS", "xQ", "xN")
+    # defaultValues <- c(xS = 950, xQ = 0.9, xN = 10000)
+    #
+    # if (missing(val)) {
+    #     val <- defaultValues[cutoffType]
+    # }
+    #
+    # # Read header and one contents line
+    # tmp <- readLines(fName, n = 2)
+    #
+    # # Parse for requested column. STRING data is " "-delimited.
+    # header <- unlist(strsplit(tmp[1], " "))
+    # iCol <- which(net == header)  # column that contains the requested net
+    #
+    # data   <- unlist(strsplit(tmp[2], " "))
+
+
+    # ==== VALIDATIONS =========================================================
+
+    # General parameter checks
+    cR <- character()
+    cR <- c(cR, .checkArgs(fMAF,        like = c("FILE_E"),   checkSize = TRUE))
+    cR <- c(cR, .checkArgs(rSNV,      like = "FILE_E",        checkSize = TRUE))
+    cR <- c(cR, .checkArgs(silent,       like = logical(1), checkSize = TRUE))
+    cR <- c(cR, .checkArgs(writeLog,     like = logical(1), checkSize = TRUE))
+
+    if(length(cR) > 0) {
+        stop(cR)
+    }
+
+    #  Validate the vector of MAF file names parameter
+    #  Validate rSNV as an existing file or create new file
+
+    # ==== READ DATA ===========================================================
+
+    # reading data will occur in multiple steps to avoid storing too much data on
+    # the disk drive and cause overload when handeling large files
+
+    # step 0: run terminal comand to gzip all folders and subfolders and remove
+    #         unwanted files like stdout, stdin, manifest etc..
+    # step 1: go through diseases for which data is available and open file
+    # step 2: read lines from file and process data as indicated in "EXTRACT" below
```

Comment:
```
Read into a temporary dataframe in one step using readr::read_delim
Read only desired columns. 
Typecheck columns after read (typecheck elements in first row). Error handling in case typecheck fails (eg. would a string in a numeric column cause the column to be converted to char? Or is readr::read_delim safe if the column type is specified?)
Process data.
Append temp data frame to old dataframe.
delete temp dataframe
explicit file close/open are unnecessary.